addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/PackageContentAccessHandler.java

```diff
@@ -0,0 +1,425 @@
+import org.commonjava.maven.galley.model.TransferOperation;
+import org.commonjava.maven.galley.transport.htcli.model.HttpExchangeMetadata;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Date;
+import java.util.function.Consumer;
+
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.formatOkResponseWithEntity;
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.formatResponse;
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.formatResponseFromMetadata;
+import static org.commonjava.indy.bind.jaxrs.util.ResponseUtils.setInfoHeaders;
+
+public class PackageContentAccessHandler
+                extends ContentAccessHandler
+{
+    public static final String CROSS = "-";
+
+    public static final String PATH_SPLITER = "/";
+
+    public static final String TGZ = ".tgz";
+
+    public static final String PACAGE_TGZ = "package" + TGZ;
+
+    @Override
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final Boolean cacheOnly, final String baseUri, final HttpServletRequest request,
+                            EventMetadata eventMetadata, final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        return doHead( packageType, type, name, packageName, null, null, cacheOnly, baseUri, request, eventMetadata,
+                       builderModifier );
+    }
+
+    @Override
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final Boolean cacheOnly, final String baseUri, final HttpServletRequest request,
+                            EventMetadata eventMetadata )
+    {
+        return doHead( packageType, type, name, packageName, null, null, cacheOnly, baseUri, request, eventMetadata,
+                       null );
+    }
+
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final String version, final String tarball, final Boolean cacheOnly, final String baseUri,
+                            final HttpServletRequest request, EventMetadata eventMetadata )
+    {
+        return doHead( packageType, type, name, packageName, version, tarball, cacheOnly, baseUri, request,
+                       eventMetadata, null );
+    }
+
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final String version, final String tarball, final Boolean cacheOnly, final String baseUri,
+                            final HttpServletRequest request, EventMetadata eventMetadata,
+                            final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        if ( !PackageTypes.contains( packageType ) )
```

Comment:
If you hard-code the package type here, you won't need to worry about fending off invalid types